[Python] Updated celery instrumentation documentation (#11600)
getsentry/sentry-docs: docs/platforms/python/integrations/celery/index.mdx

If you have the `celery` package in your dependencies, the Celery integration will be enabled automatically when you initialize the Sentry SDK.

Make sure that the **call to `sentry_sdk.init()` is loaded on worker startup**, and not only in the module where your tasks are defined. Otherwise, the initialization happens too late and events might end up not being reported.

> ### Setup Celery (Without Django)

Reviewer: ```suggestion
### Setup Celery Without Django
```

Remove parentheses to be consistent with the "Setup Celery With Django" section. Alternatively, we could rename the "Setup Celery With Django" section to "Setup Celery (With Django)."

To get the most out of Sentry make sure to initialize the Sentry SDK in your Celery worker processes as well as your application that is sending messages to Celery.

> In addition to capturing errors, you can monitor interactions between multiple services or applications by [enabling tracing](/concepts/key-terms/tracing/). You can also collect and analyze performance profiles from real users with [profiling](/product/explore/profiling/).

Reviewer 2: I agree that it's a bit jarring, but I think it's also a confusing experience to have the options to add tracing/profiling without any explanation. (Especially because we now know that a lot of people don't know what tracing is.) Maybe we could just shorten it and say: "When using Celery without Django, you'll need to initialize the Sentry SDK in both your application and the Celery worker process. 

In addition to capturing errors, you can use Sentry for [distributed tracing](/concepts/key-terms/tracing/) and [profiling](product/explore/profiling/). Select what you'd like to install to get the corresponding installation and configuration instructions below."

> In addition to capturing errors, you can monitor interactions between multiple services or applications by [enabling tracing](/concepts/key-terms/tracing/). You can also collect and analyze performance profiles from real users with [profiling](/product/explore/profiling/).

Author: Updated

> Select which Sentry features you'd like to install in addition to Error Monitoring to get the corresponding installation and configuration instructions below.

Reviewer 2: See comment above :)

> Select which Sentry features you'd like to install in addition to Error Monitoring to get the corresponding installation and configuration instructions below.

Author: updated

> #### Setup in Celery

Reviewer 2: ```suggestion
#### Set Up Celery
```

<OnboardingOptionButtons
  options={["error-monitoring", "performance", "profiling"]}
/>

```python {filename:tasks.py} {"onboardingOptions": {"performance": "12-14", "profiling": "15-18"}}
from celery import Celery, signals
import sentry_sdk

# Initializing Celery
app = Celery("tasks", broker="...")

# Initialize Sentry SDK on Celery startup

> @signals.celeryd_init.connect

Author: The both signals topic is only when using Celery Beat with Crons. This is described in the "Crons" sub page.

def init_sentry(**_kwargs):
    sentry_sdk.init(
        dsn="___PUBLIC_DSN___",
        # Set traces_sample_rate to 1.0 to capture 100%
        # of transactions for tracing.
        traces_sample_rate=1.0,
        # Set profiles_sample_rate to 1.0 to profile 100%
        # of sampled transactions.
        # We recommend adjusting this value in production.
        profiles_sample_rate=1.0,
    )

# Task definitions go here
@app.task
def add(x, y):
    return x + y
```

> The [`celeryd_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#celeryd-init) signal is triggered when the Celery deamon is started, before the worker processes are spawned. You can use the [`worker_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#worker-init) signal instead if you want to initialize Sentry on start of each worker process.

Reviewer 2: ```suggestion
The [`celeryd_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#celeryd-init) signal is triggered when the Celery deamon starts, before the worker processes are spawned. If you need to initialize Sentry for each individual worker process, us the [`worker_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#worker-init) signal instead.
```

> The [`celeryd_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#celeryd-init) signal is triggered when the Celery deamon is started, before the worker processes are spawned. You can use the [`worker_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#worker-init) signal instead if you want to initialize Sentry on start of each worker process.

Author: The both signals topic is only relevant when using Celery Beat for crons. This is described in the "Crons" subpage.

> #### Setup in Your Application

Reviewer 2: I find it confusing as well....

> #### Setup in Your Application

Author: Celery is confusing at first. 

The application and Celery stuff are completely separate. Those are two entirely separate processes you need to start: Your application and on the other hand the Celery. daemon This can also be on different machines. They are only connected through the broker (a message queue like RabbitMQ or Redis) So you have to have two Sentry.init() calls, one in your application sending messages to Celery and one in Celery processing those messages. 

People who use Celery know this and are probably not that confused. 

(Django is a special case, because you have the the Sentry config in Django and this config is used by the Django process and by the Celery process) Celery was kind of built for Django and then established itself in other setups too. 

I added some better headlines and a bit more text to reduce confusion.

<OnboardingOptionButtons
  options={["error-monitoring", "performance", "profiling"]}
/>

```python {filename:main.py} {"onboardingOptions": {"performance": "8-10", "profiling": "11-14"}}
from tasks import add
import sentry_sdk

def main():
    # Initializing Sentry SDK in our process
    sentry_sdk.init(
        dsn="___PUBLIC_DSN___",
        # Set traces_sample_rate to 1.0 to capture 100%
        # of transactions for tracing.
        traces_sample_rate=1.0,
        # Set profiles_sample_rate to 1.0 to profile 100%
        # of sampled transactions.
        # We recommend adjusting this value in production.
        profiles_sample_rate=1.0,
    )

    # Enqueueing a task to be processed by Celery
    with sentry_sdk.start_transaction(name="calling-a-celery-task"):
        result = add.delay(4, 4)

if __name__ == "__main__":
    main()
```

> ### Setup Celery With Django

Reviewer 2: ```suggestion
### Set Up Celery With Django
```

> If you're using Celery with Django in a conventional setup, have already initialized the SDK in your `settings.py` file as described in the [Django integration documentation](/platforms/python/integrations/django/#configure), and have Celery using the same settings with [`config_from_object`](https://docs.celeryq.dev/en/stable/django/first-steps-with-django.html), you don't need to initialize the SDK separately for Celery.

Reviewer 2: ```suggestion
If you're using Celery with Django in a typical setup, have initialized the SDK in your `settings.py` file (as described in the [Django integration documentation](/platforms/python/integrations/django/#configure)), and have your Celery configured to use the same settings as [`config_from_object`](https://docs.celeryq.dev/en/stable/django/first-steps-with-django.html), there's no need to initialize the Celery SDK separately.
```

## Verify

> To verify if your SDK is initialized on worker start, you can pass `debug=True` to `sentry_sdk.init()` to see extra output in your Celery logs when the SDK is initialized. If the output appears during worker startup and not only after a task has started, then it's working properly.

Reviewer: Should we perhaps add an example of how to get an error in Sentry to this section, like how we do for many of our other integrations? (potentially out of scope for this PR)

Reviewer 2: It would be great to add an example.

Author: Added an example!

> To verify if your SDK is initialized on worker start, you can pass `debug=True` to `sentry_sdk.init()` to see extra output in your Celery logs when the SDK is initialized. If the output appears during worker startup and not only after a task has started, then it's working properly.

Reviewer 2: ```suggestion
To confirm that your SDK is initialized on worker start, pass `debug=True` to `sentry_sdk.init()`. This will add extra output to your Celery logs when the SDK is initialized. If you see the output during worker startup, and not just after a task has started, then it's working correctly.
```
